Bug: incorrect peer dependencies in v8.0.0-alpha.10 packages

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Playground Link

_No response_

### Repro Code

```TypeScript
npm init
npm add --save-dev eslint@v9.2.0
npm add --save-dev typescript-eslint@v8.0.0-alpha.10
npm i
```


### ESLint Config

```javascript
N/A
```


### tsconfig

```jsonc
N/A
```


### Expected Result

I expected that depending on the 8.0.0-alpha.10 version of typescript-eslint would exclusively use 8.0.0-alpha.10 releases of packages, and thus only depend on eslint v9.

### Actual Result

While installation succeeds, there is a warning about an invalid peer dependency, and it shows that typescript-eslint's v8 alpha packages and v7 packages are being commingled.

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @typescript-eslint/parser@7.8.0
npm WARN Found: eslint@9.2.0
npm WARN node_modules/eslint
npm WARN   dev eslint@"^9.2.0" from the root project
npm WARN   4 more (@eslint-community/eslint-utils, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer eslint@"^8.56.0" from @typescript-eslint/parser@7.8.0
npm WARN node_modules/@typescript-eslint/parser
npm WARN   peer @typescript-eslint/parser@"^7.0.0" from @typescript-eslint/eslint-plugin@8.0.0-alpha.10
npm WARN   node_modules/@typescript-eslint/eslint-plugin
npm WARN 
npm WARN Conflicting peer dependency: eslint@8.57.0
npm WARN node_modules/eslint
npm WARN   peer eslint@"^8.56.0" from @typescript-eslint/parser@7.8.0
npm WARN   node_modules/@typescript-eslint/parser
npm WARN     peer @typescript-eslint/parser@"^7.0.0" from @typescript-eslint/eslint-plugin@8.0.0-alpha.10
npm WARN     node_modules/@typescript-eslint/eslint-plugin
```

### Additional Info

The problem first shows up in the peer dependencies of the eslint-plugin package, but I cannot select that package in the dropdown.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.0.0-alpha.10` |
| `@typescript-eslint/parser`            | `7.8.0` |
| `@typescript-eslint/scope-manager`     | both `7.8.0` and `8.0.0-alpha.10` |
| `@typescript-eslint/typescript-estree` | both `7.8.0` and `8.0.0-alpha.10` |
| `@typescript-eslint/type-utils`        | `8.0.0-alpha.10` |
| `@typescript-eslint/utils`             | `8.0.0-alpha.10` |
| `TypeScript`                           | `5.4.5` |
| `ESLint`                               | `9.2.0` |
| `node`                                 | `20.12.2` |

--- a/packages/eslint-plugin/package.json
+++ b/packages/eslint-plugin/package.json
@@ -102,7 +102,7 @@
     "unist-util-visit": "^5.0.0"
   },
   "peerDependencies": {
-    "@typescript-eslint/parser": "^7.0.0",
+    "@typescript-eslint/parser": "^8.0.0",
     "eslint": "^8.57.0 || ^9.0.0"
   },
   "peerDependenciesMeta": {

